src/components/document-list/_macro.spec.js

```diff
@@ -4,472 +4,359 @@ import * as cheerio from 'cheerio';
+                    }),
+                );
+                expect($('#some-id').length).toBe(1);
+            });
         });
+    });
 
-        it('has the correct container if `fullWidth`', () => {
-            const $ = cheerio.load(
-                renderComponent('document-list', {
-                    documents: [{ ...EXAMPLE_DOCUMENT_LIST_BASIC, featured: true, fullWidth: true }],
-                }),
-            );
-
-            expect($('.ons-container').length).toBe(1);
+    describe('GIVEN: Params: custom-class', () => {
+        describe('WHEN: custom class is provided', () => {
+            test('THEN: applies the right custom classes', () => {
+                const $ = cheerio.load(
+                    renderComponent('document-list', {
+                        classes: 'custom-class',
+                        documents: [EXAMPLE_DOCUMENT_LIST_BASIC, EXAMPLE_DOCUMENT_LIST_BASIC, EXAMPLE_DOCUMENT_LIST_BASIC],
+                    }),
+                );
+                expect($('.ons-document-list').hasClass('custom-class')).toBe(true);
+            });
         });
+    });
 
-        it('has the correct container class if `fullWidth` and `wide`', () => {
-            const $ = cheerio.load(
-                renderComponent('document-list', {
-                    documents: [{ ...EXAMPLE_DOCUMENT_LIST_BASIC, featured: true, fullWidth: true, wide: true }],
-                }),
-            );
+    describe('GIVEN: Params: featured', () => {
+        describe('WHEN: featured is set for a document', () => {
+            test('THEN: applies the featured class to the document item', () => {
+                const $ = cheerio.load(
+                    renderComponent('document-list', {
+                        documents: [{ ...EXAMPLE_DOCUMENT_LIST_BASIC, featured: true }],
+                    }),
+                );
 
-            expect($('.ons-container--wide').length).toBe(1);
+                expect($('.ons-document-list__item--featured').length).toBe(1);
+            });
         });
+    });
 
-        it('has the correct container class if `featured`', () => {
-            const $ = cheerio.load(
-                renderComponent('document-list', {
-                    documents: [{ ...EXAMPLE_DOCUMENT_LIST_BASIC, featured: true }],
-                }),
-            );
+    describe('GIVEN: Params: fullWidth', () => {
+        describe('WHEN: fullWidth is set for a document with basic parameters', () => {
+            test('THEN: does not apply container class', () => {
+                const $ = cheerio.load(
+                    renderComponent('document-list', {
+                        documents: [{ ...EXAMPLE_DOCUMENT_LIST_BASIC, fullWidth: true }],
+                    }),
+                );
 
-            expect($('.ons-document-list__item--featured').length).toBe(1);
+                expect($('.ons-container').length).toBe(0);
```

Comment:
Think this needs to be a bit more specific about where it is checking for the container class which would be inside the `ons-document-list__item`. Same with the other tests like this